### PR TITLE
Small changes to post-request.

### PR DESCRIPTION
**Pull request by farzaa**

### CODE REVIEW

File: server/gateway/handlers/code.go [hunk 1]
```diff
@@ -13,6 +13,8 @@ import (
 
 type Submission struct {
 	Code string `json:"code"`
+	Lesson int `json:"lesson"`
+	Step int `json:"step"`
```
Comment (evanfrawley) [inline server/gateway/handlers/code.go]: in Goland, hit `ctl + opt + L` to reformat the code, it'll align everything nicely here, which is Go preferred style ;)

File: server/gateway/handlers/code.go [hunk 2]
```diff
@@ -30,6 +32,7 @@ func (ctx *HandlerContext) CodeHandler(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
+		fmt.Println(c)
```
Comment (evanfrawley) [inline server/gateway/handlers/code.go]: do we need to print this here?

File: server/gateway/runners/python.go [hunk 3]
```diff
@@ -72,6 +72,30 @@ func WaitOnContainer(cli *client.Client, ctx context.Context, id string) (bool)
 	return false
 }
 
+func ReadPythonContainerResult(closer io.ReadCloser)(*bytes.Buffer, error) {
```
Comment (evanfrawley) [inline server/gateway/runners/python.go]: reusability Kreygasm

File: server/gateway/runners/python.go [hunk 4]
```diff
@@ -72,6 +72,30 @@ func WaitOnContainer(cli *client.Client, ctx context.Context, id string) (bool)
...
+	tr := tar.NewReader(closer)
+	for {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			break // End of archive
+		}
+		if err != nil {
+			return nil, err
+		}
+		log.Printf("Contents of %s:\n", hdr.Name)
```
Comment (evanfrawley) [inline server/gateway/runners/python.go]: do these println's need to be here?

File: server/gateway/runners/python.go [hunk 5]
```diff
@@ -125,28 +149,9 @@ func SendToPythonContainer(filename string) (*bytes.Buffer, error) {
...
-			return nil, err
-		}
-		fmt.Println()
-	}
-	// Close the reader that the Docker API created.
-	copyReader.Close()
-
-	log.Println(buff.String())
-	return buff, cli.ContainerRemove(ctx, resp.ID, types.ContainerRemoveOptions{Force: true})
+	buffer, err := ReadPythonContainerResult(copyReader)
```
Comment (evanfrawley) [inline server/gateway/runners/python.go]: i think we'll still need to add an error check here to handle the case that reading isn't good 🤷‍♂️

**Comment by evanfrawley** [comment]
TLDR, remove lotsa printlns, use nifty go style helper, and do 1 more error check
